src/internal/cm/abi.go

```diff
@@ -84,6 +89,25 @@ func F64ToU64(v float64) uint64 { return *(*uint64)(unsafe.Pointer(&v)) }
 // [Canonical ABI]: https://github.com/WebAssembly/component-model/blob/main/design/mvp/CanonicalABI.md
 func U64ToF64(v uint64) float64 { return *(*float64)(unsafe.Pointer(&v)) }
 
+// F32ToU64 maps the bits of a [float32] into a [uint64].
+// Used to lower a [float32] into a Core WebAssembly i64 when required by the [Canonical ABI].
+//
+// [float32]: https://pkg.go.dev/builtin#float32
+// [uint64]: https://pkg.go.dev/builtin#uint64
+// [Canonical ABI]: https://github.com/WebAssembly/component-model/blob/main/design/mvp/CanonicalABI.md
+func F32ToU64(v float32) uint64 { return uint64(*(*uint32)(unsafe.Pointer(&v))) }
+
+// U64ToF32 maps the bits of a [uint64] into a [float32].
+// Used to lift a Core WebAssembly i64 into a [float32] when required by the [Canonical ABI].
+//
+// [uint64]: https://pkg.go.dev/builtin#uint64
+// [float32]: https://pkg.go.dev/builtin#float32
+// [Canonical ABI]: https://github.com/WebAssembly/component-model/blob/main/design/mvp/CanonicalABI.md
+func U64ToF32(v uint64) float32 {
```

Comment:
This comment should probably apply to the upstream generator, but note that `math` already has
```
[func Float32bits(f float32) uint32](https://pkg.go.dev/math#Float32bits)
[func Float32frombits(b uint32) float32](https://pkg.go.dev/math#Float32frombits)
[func Float64bits(f float64) uint64](https://pkg.go.dev/math#Float64bits)
[func Float64frombits(b uint64) float64](https://pkg.go.dev/math#Float64frombits)
```

Reply:
Memorializing conversation from IRL this past week in San Francisco:

"We duplicate functionality from package `math` here to avoid a potential circular dependency in package `runtime`. Package `cm` intentionally depends only on `unsafe`."